nodejs/lancedb/arrow.ts

```diff
@@ -375,12 +376,71 @@ function makeVector(
   }
 }
 
+async function appendVectorColumn(
+  table: ArrowTable,
+  schema: Schema,
+): Promise<ArrowTable> {
+  const registry = getRegistry();
+  const functions = registry.parseFunctions(schema.metadata);
+
+  const columns = Object.fromEntries(
+    table.schema.fields.map((field) => [
+      field.name,
+      table.getChild(field.name)!,
+    ]),
+  );
+
+  for (const functionEntry of functions.values()) {
+    const sourceColumn = columns[functionEntry.sourceColumn];
+    const destColumn = functionEntry.vectorColumn ?? "vector";
+    if (sourceColumn === undefined) {
```

Comment:
This is just for my own knowledge but how do you decide between `sourceColumn === undefined` and `sourceColumn === undefined || sourceColumn === null` in these validation checks?  (i.e. how do you know when you only need to check for one and not both)

Reply:
so `null` usually has to be explicitly set, while `undefined` is just the absence of a value. So as long as we never explicitly start setting values to null, we generally only need to check for `undefined`.